pkg/agent/agent.go

```diff
@@ -785,7 +798,7 @@ func (i *Initializer) initNodeLocalConfig() error {
 
 // initializeIPSec checks if preconditions are met for using IPsec and reads the IPsec PSK value.
 func (i *Initializer) initializeIPSec() error {
-	if !i.networkConfig.EnableIPSecTunnel {
+	if i.networkConfig.TrafficEncryptionMode != config.TrafficEncryptionModeIPSec {
```

Comment:
I feel better to check this at the caller by calling initializeIPSec(). If the intention is to check and clean up in initializeIPSec() later, then add a TODO comment before initializeIPSec()